### ISSUE TITLE
Large archives are not supported

### ISSUE THREAD

**Issue by halindrome** [redhotpenguin/perl-Archive-Zip#18]
Unfortunately this module does not support "zip64".  I imagine supporting it could be pretty difficult, but another module (IO::Compress::Zip) seems to.  Maybe there is a way to leverage that?

**Comment by redhotpenguin** [comment]
Yeah I've looked into Zip64 support at the request of the US geological
survey users. It is not trivial. I could call out to IO::Compress::Zip, but
I think we'd lose the file member functionality (and also the performance).

**Comment by halindrome** [comment]
I think you are right on both counts.  But I don't know how different the
zip64 signatures are from the basic zip signatures.

On Thu, Aug 13, 2015 at 9:48 PM, Fred Moyer <notifications@github.com>
wrote:

> Yeah I've looked into Zip64 support at the request of the US geological
> survey users. It is not trivial. I could call out to IO::Compress::Zip, but
> I think we'd lose the file member functionality (and also the performance).
>
> On Thu, Aug 13, 2015 at 6:19 AM, Shane McCarron <notifications@github.com>
> wrote:
>
> > Unfortunately this module does not support "zip64". I imagine supporting
> > it could be pretty difficult, but another module (IO::Compress::Zip)
> seems
> > to. Maybe there is a way to leverage that?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/redhotpenguin/perl-Archive-Zip/issues/18>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/redhotpenguin/perl-Archive-Zip/issues/18#issuecomment-130934104>
> .
>



-- 
Shane McCarron
halindrome@gmail.com

**Comment by pmqs** [comment]
See PR #45

**Comment by redhotpenguin** [comment]
I'm pushing 1.66 with the zip64 changes tonight. So far all tests pass, but it may need an iteration.

**Comment by pmqs** [comment]
@redhotpenguin - on Windows  I'm seeing #45 hanging the appveyor CI job in **t/26_bug_26.t** (see [here](https://ci.appveyor.com/project/pmqs/perl-archive-zip/builds/27458396)) . On Linux the same test just fails. Might be an idea to get an appveyor account setup?

**Comment by pmqs** [comment]
There are a lot of failures in cpantesters (see [here](http://cpantesters.org/distro/A/Archive-Zip.html))

There appear to be  two main issues

First is the use of "**Q**" in unpack. That is only available in a 64-bit Perl build. 

```
#   Failed test at t/03_ex.t line 38.
#          got: '512'
#     expected: '0'
Invalid type 'Q' in unpack at /var/tmp/build/Archive-Zip-1.66-22/blib/lib/Archive/Zip/Archive.pm line 880.
```

The next issue is the use of the "**//**" operator. This needs perl 5.10.0 or better. I think this is the reason for all the 5.8.x failure in cpantesters.

**Comment by redhotpenguin** [comment]
I'll get appveyor setup and see if I can resolve the pack issue, have some
tuits today

**Comment by pmqs** [comment]
@farblos - just highlighting that there are issues with the Zip64 PR in case you haven't seen this thread.

**Comment by farblos** [comment]
@pmqs - Thanks for highlighting ..

As for possible solutions:

'Q': no easy-to-implement idea here so I'm counting on Fred
'//': should be easily solvable but there might be other five-ten-isms lurking since that is what I have been working with

Please let me know if you expect me to provide fixes.

**Comment by farblos** [comment]
Just opened unrelated zip64 issue #51 which ideally should also be fixed with above issues ...

**Comment by pmqs** [comment]
Created #52 to (hopefully) address the `//` issue.

**Comment by redhotpenguin** [comment]
Closing, fixed in 1.67
